### ISSUE TITLE
Use "Additional URLs" fields for Auto-Type feature

### ISSUE THREAD

**Issue by Hidigoudi** [keepassxreboot/keepassxc#10991]
## Summary
Add the possibility to use the "Additional URLs" fields when only using the Global Auto-Type feature.

## Examples
When the browser integration is disabled, we can not add additional URLs :
![image](https://github.com/keepassxreboot/keepassxc/assets/173546307/2c0c2924-f999-4893-82e1-2828fa450d3e)

## Context
Currently, it is required to enable "Browser Integration" option in general settings to configure several URLs per entry. When enabling this option and use Auto-Type, it does not search in all additional URLs but only in the main "URL" field. It could be very convenient to add this feature for Auto-Type users.

**Comment by leumasme** [comment]
The wording is a bit confusing here, it seems like you're describing two issues
- Browser Integration 'Additional URLs' are not used for Auto-Type search/detection
  - For Search: The Normal search also doesn't search additional URLs/attributes, so this is consistent, although suboptimal. Searching `attr:<your url>` will include attributes in search
- Additional URLs are configured via the "Browser Integration" tab, which is not shown when Browser Integration is disabled
  - From what I know, the Additional URLs are just Additional Attributes ("Advanced" tab) with the name `KP2A_URL_N` and the URL in the value
  - This is only an issue if the Browser Integration attributes are treated as special attributes even outside of Browser Integration
 
Auto-Type detection using addional URLs is clearly useful.
If the Auto-Type search then doesn't also include additional URLs by default, it's a bit inconsistent.
If the Auto-Type search does include additional URLs but the normal search does not, it's again inconsistent.
Treating the Browser Integration URL attributes as special magic attributes for Auto-Type detection is perhaps also a bit suboptimal when one wants to use Auto-Type with secondary URLs but Browser Integration disabled, as the original issue describes.
IMO Unclear what exactly the expected behavior should be here.

**Comment by Hidigoudi** [comment]
The thing is that I don't know why the additional URLs feature is linked to the Browser Integration, I don't see the connection between both.

For me it should be implemented by default and especially for people that use only Auto-Type instead of the keepassxc web extension.

Currently if I force the behavior and enable the browser integration option (even if I do not have the extension) and add additional URLs in an entry, the Auto-Type does not fill and does not found the additional URLs.

**Comment by droidmonkey** [comment]
> The thing is that I don't know why the additional URLs feature is linked to the Browser Integration, I don't see the connection between both

Quite simply because the only feature that currently uses additional urls is the browser feature...

Auto-type is primarily meant to be used with window title matching, not url matching.

**Comment by Hidigoudi** [comment]
> > The thing is that I don't know why the additional URLs feature is linked to the Browser Integration, I don't see the connection between both
> 
> Quite simply because the only feature that currently uses additional urls is the browser feature...
> 
> Auto-type is primarily meant to be used with window title matching, not url matching. 

True, it is one (and probably the default) possibility in options : only use the Window title matching to fill fields. However the software (Auto-Type) allows you to rely solely on a match of a URL instead of titles. I know and I admit it is (probably?) a rare configuration but it exists.

By the way, it is not very annoying and illogical to have to enable the Browser integration option to get additional URLs panel. The only annoying thing is that Auto-Type does not use it. 

I think it is maybe a "compromise" to avoid too much work on this. Only make Auto-Type search also in the additional URLs fields.

**Comment by varjolintu** [comment]
This should also include support for the new URL wildcards with Additional URLs.
